libsel4/include/sel4/constants.h

```diff
@@ -83,6 +83,16 @@ typedef enum {
 /* the size of a single extra refill */
 #define seL4_RefillSizeBytes (2 * 8)
 
+/* Reasons for why a timeout fault occurred */
+enum {
+    /* The thread was scheduled without an SC */
```

Comment:
Are seL4_Timeout_NoSC and seL4_Timeout_Unconfigured valuable as seperate values? Could this just be seL4_Timeout_InvalidSC??

Reply:
One is plausible in the case that a client may have performed a send that refused to donate an SC, the other means that you are probably doing SC configuration in a questionable manner (but in a way that the kernel doesn't statically prevent) and is probably more useful as a debugging diagnostic than an actual fault case that would show up in a sane system.

Reply:
Lol, seL4 doesn't have debugging diagnostics! 
In all seriousness though, we do want to keep the code small, so in general we give out as little info as possible in a release kernel ...

Reply:
I honestly wasn't sure whether to present these differently. I figured that we distinguish the cases as the actual faulting circumstances are discernible even if no real-world system would actually care about the distinction (much like many hardware fault details I guess). It also means we can write tests that more directly show that the kernel behaves as we want in these circumstances, but that in itself is not a great motivator.